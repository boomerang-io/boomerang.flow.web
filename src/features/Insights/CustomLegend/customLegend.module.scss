@import "~Styles/variables";

.container {
  display: flex;
  justify-content: center;
  width: 100%;
  height: 2rem;
  margin-top: 2.5rem;
}
.iconContainer {
  display: flex;
  align-items: center;
  cursor: pointer;
}
.icon {
  background-color: transparent;
  width: 1.875rem;
  height: 1rem;
  color: $dark-gray;
}
.label {
  font-size: 0.875rem;
  margin-right: 1rem;
  &.--toggled {
    opacity: 0.4;
  }
}
