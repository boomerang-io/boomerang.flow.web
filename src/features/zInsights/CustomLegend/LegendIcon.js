import React from "react";
import PropTypes from "prop-types";

const LegendIcon = ({ className, strokeColor }) => {
  return (
    <svg
      className={className}
      width="100%"
      height="100%"
      viewBox="10 0 13 28"
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        stroke-width="4"
        fill="none"
        stroke={strokeColor}
        d="M0,16h10.666666666666666
          A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
          H32M21.333333333333332,16
          A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
        class="recharts-legend-icon"
      />
    </svg>
  );
};

LegendIcon.propTypes = {
  className: PropTypes.string,
  strokeColor: PropTypes.string
};

export default LegendIcon;
