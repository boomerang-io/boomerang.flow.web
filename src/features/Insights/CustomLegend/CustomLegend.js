import React, { Component } from "react";
import PropTypes from "prop-types";
import LegendIcon from "./LegendIcon";
import isAccessibleEvent from "@boomerang/boomerang-utilities/lib/isAccessibleEvent";
import styles from "./customLegend.module.scss";
class CustomTooltip extends Component {
  render() {
    // const tooltipFields = this.props.payload.length > 0 ? Object.keys(this.props.payload[0].payload) : null;

    return (
      <div className={styles.container}>
        {this.props.payload.map((data, index) => {
          let isToggled = this.props.toggledItems.find(item => data.payload.name === item);
          return (
            <div
              key={`${data.payload.name}-${index}`}
              className={styles.iconContainer}
              onMouseEnter={this.props.onMouseEnter}
              onMouseLeave={this.props.onMouseLeave}
              onClick={() => this.props.toggleItem(data)}
              onKeyDown={e => isAccessibleEvent(e) && this.props.toggleItem(data)}
              role="button"
              tabIndex="0"
            >
              <LegendIcon className={styles.icon} strokeColor={data.payload.fill} />
              <span className={`${styles.label} ${isToggled ? " --toggled" : ""}`} key={data.dataKey}>
                {data.payload.name}
              </span>
            </div>
          );
        })}
      </div>
    );
  }
}

CustomTooltip.propTypes = {
  payload: PropTypes.array,
  toggleItem: PropTypes.func,
  toggledItems: PropTypes.array,
  onMouseEnter: PropTypes.func,
  onMouseLeave: PropTypes.func
};

export default CustomTooltip;
