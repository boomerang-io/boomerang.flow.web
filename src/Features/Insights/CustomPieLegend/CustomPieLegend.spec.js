import React from "react";
import CustomLegend from "./index";

const payload = [
  {
    color: "#047cc0",
    dataKey: "total",
    fill: "url(#blueGradient)",
    fillOpacity: 0.6,
    formatter: undefined,
    name: "total",
    payload: { date: 1542160248412, failed: 5, success: 12, total: 17 },
    stroke: "#047cc0",
    unit: undefined,
    value: 17,
  },
];
const mockfn = jest.fn();
const toggleItem = mockfn;
const toggledItems = [];

describe("CustomLegend --- Snapshot", () => {
  it("Capturing Snapshot of CustomLegend", () => {
    const { baseElement } = rtlContextRouterRender(
      <CustomLegend payload={payload} toggleItem={toggleItem} toggledItems={toggledItems} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
