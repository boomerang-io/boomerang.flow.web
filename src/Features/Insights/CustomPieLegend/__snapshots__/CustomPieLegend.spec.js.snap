// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomLegend --- Snapshot Capturing Snapshot of CustomLegend 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="container"
    >
      <div
        class="iconContainer"
        role="button"
        tabindex="0"
      >
        <div
          class="icon"
        />
        <div
          class="labelContainer"
        >
          <span
            class="label"
          />
          <span
            class="value"
          >
            undefined% (undefined runs)
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
